service/internal/graph/nodes.go

```diff
@@ -253,6 +253,10 @@ func (n *connectorNode) buildComponent(
 				return err
 			}
 			n.Component = conn
+
+			// The connector may mutate data itself
```

Comment:
Very interesting that you do this only for traces (same for the capability consumer initialization). Is this not a bug? Can we also start extracting this in a separate func? (hard to follow)

Reply:
I'll more comments to clarify this, but note the comment above the next line:
```
// When connecting pipelines of the same data type, the connector must
// inherit the capabilities of pipelines in which it is acting as a receiver.
n.baseConsumer = capabilityconsumer.NewTraces(conn, capability)
```

What's going on here is that when we connect pipelines of the same type, the original data may be mutated later on. However, when we are _not_ connecting pipelines of the same type, the connector does not pass the original data along, so there is no way the data is mutated downstream.

So in the different-type cases, we already correctly represent capabilities by setting `n.baseConsumer` as the connector.

But in the same-type cases we are building a new `capabilityconsumer`, but were forgetting to include the capability of the connector itself.

Reply:
I moved this line and combined the comment with the one below, since they are so closely related.